packages/shared_preferences/shared_preferences/android/src/main/java/io/flutter/plugins/sharedpreferences/MethodCallHandlerImpl.java

```diff
@@ -118,17 +121,23 @@ public void onMethodCall(MethodCall call, MethodChannel.Result result) {
 
   private void commitAsync(
       final SharedPreferences.Editor editor, final MethodChannel.Result result) {
-    new AsyncTask<Void, Void, Boolean>() {
-      @Override
-      protected Boolean doInBackground(Void... voids) {
-        return editor.commit();
-      }
+    final ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
might it be more beneficial to use a cached thread pool here with minimal size of 0 and maximal of 1, so that there is no threads allocated if shared preferences weren't used for a while?

```
new ThreadPoolExecutor(0, 1,
                                  30L, TimeUnit.SECONDS,
                                  new SynchronousQueue<Runnable>());
```

Reply:
That makes sense 🤔